src/lib/mina_base/parties_logic.ml

```diff
@@ -817,6 +838,48 @@ module Make (Inputs : Inputs_intf) = struct
       let a = Account.set_timing timing a in
       (a, local_state)
     in
+    (* Transform into a snapp account.
+       This must be done before updating snapp fields!
+    *)
+    let a = Account.make_snapp a in
+    (* Update app state. *)
+    let a, local_state =
+      let app_state = Party.Update.app_state party in
+      let keeping_app_state =
+        Bool.all
+          (List.map ~f:Set_or_keep.is_keep
+             (Pickles_types.Vector.to_list app_state))
+      in
+      let changing_app_state =
+        Bool.all
```

Comment:
``` 
        Bool.any
          (List.map ~f:Set_or_keep.is_set
             (Pickles_types.Vector.to_list app_state))
```

Reply:
Copy paste error, good catch :man_facepalming: The name here is perhaps unclear, it should be
```ocaml
let changing_entire_app_state =
  Bool.all
    (List.map ~f:Set_or_keep.is_set
      (Pickles_types.Vector.to_list app_state))
```
so that the account's `proved_state` boolean is only true if a proof was provided that set the entire state (initialisation) and then the state was only subsequently updated by proofs (proved transitions).

Reply:
should we rename it to maybe `proved_state_reset` or something?